discord/ui/container.py

```diff
@@ -0,0 +1,415 @@
+
+    @property
+    def width(self):
+        return 5
+
+    def _is_v2(self) -> bool:
+        return True
+
+    def to_components(self) -> List[Dict[str, Any]]:
+        components = []
+        for child in sorted(self._children, key=lambda i: i._rendered_row or 0):
+            components.append(child.to_component_dict())
+        return components
+
+    def to_component_dict(self) -> Dict[str, Any]:
+        components = self.to_components()
+
+        colour = None
+        if self._colour:
+            colour = self._colour if isinstance(self._colour, int) else self._colour.value
+
+        base = {
+            'type': self.type.value,
+            'accent_color': colour,
+            'spoiler': self.spoiler,
+            'components': components,
+        }
+        if self.id is not None:
+            base['id'] = self.id
+        return base
+
+    def _update_store_data(
+        self,
+        dispatch_info: Dict[Tuple[int, str], Item[Any]],
+        dynamic_items: Dict[Any, Type[DynamicItem]],
+    ) -> bool:
+        is_fully_dynamic = True
+        for item in self.__dispatchable:
+            if isinstance(item, DynamicItem):
+                pattern = item.__discord_ui_compiled_template__
+                dynamic_items[pattern] = item.__class__
+            elif item.is_dispatchable():
+                dispatch_info[(item.type.value, item.custom_id)] = item
+                is_fully_dynamic = False
+        return is_fully_dynamic
+
+    @classmethod
+    def from_component(cls, component: ContainerComponent) -> Self:
+        return cls(
+            *[_component_to_item(c) for c in component.children],
+            accent_colour=component.accent_colour,
+            spoiler=component.spoiler,
+            id=component.id,
+        )
+
+    def walk_children(self) -> Generator[Item[V], None, None]:
+        """An iterator that recursively walks through all the children of this container
+        and it's children, if applicable.
+
+        Yields
+        ------
+        :class:`Item`
+            An item in the container.
+        """
+
+        for child in self.children:
+            yield child
+
+            if getattr(child, '__discord_ui_update_view__', False):
+                # if it has this attribute then it can contain children
+                yield from child.walk_children()  # type: ignore
+
+    def add_item(self, item: Item[Any]) -> Self:
+        """Adds an item to this container.
+
+        This function returns the class instance to allow for fluent-style
+        chaining.
+
+        Parameters
+        ----------
+        item: :class:`Item`
+            The item to append.
+
+        Raises
+        ------
+        TypeError
+            An :class:`Item` was not passed.
+        ValueError
+            Maximum number of children has been exceeded (10).
+        """
+
+        if len(self._children) >= 10:
+            raise ValueError('maximum number of children exceeded')
```

Comment:
This is no longer a limit: https://discord.com/channels/613425648685547541/697138785317814292/1367197980230156478